tests/src/test/java/alluxio/client/cli/JournalToolTest.java

```diff
@@ -254,10 +256,28 @@ private long getCurrentCopyCatSnapshotIndex(String journalFolder) throws Throwab
     return currentSnapshotIdx.get();
   }
 
-  private String findCheckpointDir() throws IOException {
-    List<Path> checkpoint = Files.list(mDumpDir.toPath())
-        .filter(p -> p.toString().contains("checkpoints-")).collect(toList());
-    assertEquals("Unexpected checkpoint list: " + checkpoint, 1, checkpoint.size());
+  private String findCheckpointDir() throws Exception {
+    final List<Path> checkpoint = new ArrayList<>();
+    CommonUtils.waitFor("Checkpoint list:", () -> {
+      List<Path> checkpointTest = null;
+      try {
+        checkpointTest = Files.list(mDumpDir.toPath())
+            .filter(p -> p.toString().contains("checkpoints-")).collect(toList());
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+
+      if (checkpointTest.size() == 0) {
+        return false;
+      }
+      checkpoint.addAll(checkpointTest);
+      return true;
+    }, WaitForOptions.defaults().setTimeoutMs(500));
+
+    if (checkpoint.size() > 1) {
```

Comment:
Can we do an `assertEquals(1, checkpoint.size());` instead? Or, we can just use the previous code assert line?

Reply:
Oh duh yeah you are right.